js/background.js

```diff
@@ -498,11 +498,17 @@
             }
 
             return wrapDeferred(conversation.save(updates)).then(function() {
-                if (typeof details.expireTimer !== 'undefined') {
+                const { expireTimer } = details;
+                const hasExpireTimer = typeof expireTimer !== 'undefined';
+                // TODO: Remove once
+                // https://github.com/signalapp/Signal-Desktop/issues/2079
+                // is resolved:
+                const isInvalidIOSExpireTimerReset = expireTimer === null;
```

Comment:
Does this work? 

My understanding of the iOS bug was not that we set the field with a weird value, but that we just never set it.

Reply:
Here's a link to the diff: 

https://github.com/signalapp/Signal-iOS/pull/3075/files